tests/main/proxy/task.yaml

```diff
@@ -0,0 +1,28 @@
+summary: Ensure that the core.proxy.* settings are honored
+
+systems: [ubuntu-18.04-64]
+
+restore: |
+    snap set core proxy.https=
+    apt autoremove --purge -y squid
+
+prepare: |
+    rm -rf /var/log/squid/*
+    apt install -y squid
+
+execute: |
+    echo "Setup proxy config"
+    snap set core proxy.https=http://localhost:3128
+
+    echo "Check that the commands go through the proxy"
+    snap find test-snapd-tools | MATCH test-snapd-tools
+
+    # ensure squid wrote the access.log
+    systemctl restart squid
+    for _ in $(seq 10); do
+        if grep api.snapcraft.io /var/log/squid/access.log; then
+            break
+        fi
+        sleep 1h
```

Comment:
Wait, _one hour_?!?

Reply:
*cough* sorry, that was a typo. Good thing we have reviews (I think I added this late at night, should not do that!)